prac_09/taxi_simulator.py

```diff
@@ -0,0 +1,59 @@
+"""Prac 09 - Taxi Simulator"""
+
+from prac_09.silver_service_taxi import SilverServiceTaxi
+from prac_09.taxi import Taxi
+
+MENU = """q)uit, c)hoose taxi, d)rive"""
+
+
+def main():
+    """Execute Taxi Simulator"""
+    current_taxi = None
+    bill_to_date = 0
+    taxis = [Taxi("Prius", 100), SilverServiceTaxi("Limo", 100, 2), SilverServiceTaxi("Hummer", 200, 4)]
+    print("Let's drive!")
+    print(MENU)
+    choice = input(">>> ").lower().strip()
+    while choice != "q":
+        bill = 0
+        if choice == "c":
+            current_taxi = choose_taxis(taxis)
+        elif choice == "d":
+            bill = determine_cost(current_taxi, taxis)
+        else:
+            print("Invalid choice")
+        bill_to_date += bill
+        print(f"Bill to date: {bill_to_date}")
+        print(MENU)
+        choice = input(">>> ").lower().strip()
+
+
+def determine_cost(current_taxi, taxis):
+    """Determines the cost of a taxi"""
+    try:
```

Comment:
I had a different way of checking where it loops, and if the person gives a non integer it says "invalid input" and if they give a distance of < 0, it says invalid distance. This just reduced the amount of error check statements then does the calculations after getting the value which (in my opinion) is a nicer way of doing things. That of course doesn't mean it's more valid than yours though.